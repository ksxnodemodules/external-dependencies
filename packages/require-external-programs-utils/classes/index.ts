import Grouper from './grouper'
export { Grouper }
