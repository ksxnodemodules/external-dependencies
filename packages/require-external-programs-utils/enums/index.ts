export enum ExitStatusCodes {
  Satisfied = 0,
  Unsatisfied = 1,
  Exception = 2
}
