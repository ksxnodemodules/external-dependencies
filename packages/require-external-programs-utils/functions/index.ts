import group from './group'
import groupByManifest from './group-by-manifest'
import groupByProgram from './group-by-program'
import messageFromGrouper from './message-from-grouper'
import messageFromCheckerResult from './message-from-checker-result'
import examine from './examine'

export {
  group,
  groupByManifest,
  groupByProgram,
  messageFromGrouper,
  messageFromCheckerResult,
  examine
}
