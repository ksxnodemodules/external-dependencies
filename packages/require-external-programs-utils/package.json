{
  "name": "require-external-programs-utils",
  "description": "Utilities used by require-external-programs package",
  "version": "0.0.1",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/external-dependencies#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/external-dependencies.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/external-dependencies/issues"
  },
  "keywords": [
    "utility",
    "external",
    "programs",
    "require"
  ],
  "dependencies": {
    "chalk": "^2.4.1",
    "advanced-map-initialized": "^0.0.2",
    "tslib": "^1.9.3",
    "require-external-programs-lib": "^0.0.1",
    "require-external-programs-db": "^0.0.1",
    "@types/which": "^1.3.1",
    "@types/node": "^10.12.0"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
