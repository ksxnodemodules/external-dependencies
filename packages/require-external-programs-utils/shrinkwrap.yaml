dependencies:
  '@types/node': 10.12.18
  '@types/which': 1.3.1
  advanced-map-initialized: 0.0.2
  chalk: 2.4.1
  require-external-programs-db: 'link:../require-external-programs-db'
  require-external-programs-lib: 'link:../require-external-programs-lib'
  tslib: 1.9.3
packages:
  /@types/node/10.12.18:
    dev: false
    resolution:
      integrity: sha512-fh+pAqt4xRzPfqA6eh3Z2y6fyZavRIumvjhaCL753+TVkGKGhpPeyrJG2JftD0T9q4GF00KjefsQ+PQNDdWQaQ==
  /@types/which/1.3.1:
    dev: false
    resolution:
      integrity: sha512-ZrJDWpvg75LTGX4XwuneY9s6bF3OeZcGTpoGh3zDV9ytzcHMFsRrMIaLBRJZQMBoGyKs6unBQfVdrLZiYfb1zQ==
  /advanced-map-base/0.0.2:
    dependencies:
      '@types/node': 10.12.18
      advanced-map-types: 0.0.2
      tslib: 1.9.3
      typescript-miscellaneous: 0.1.1
    deprecated: Depends on wrong version of "convenient-typescript-utilities"
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-fj4I4dNaEnVl9bSKsMXtgyVo4+kgV04ScFyLjyIfdNWp1FnRsCzWHtZDGblgq/Nau/2EMegOqW+PDN9qKmGDIA==
  /advanced-map-initialized/0.0.2:
    dependencies:
      '@types/node': 10.12.18
      advanced-map-base: 0.0.2
      advanced-map-types: 0.0.2
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-L6/yZylpOElrSpsZTPiMvpCsgnybX8JxYZX88EO2h+yNHYoqwRCojjNOifKSuoIFiWUb5mPFA0tlW7oTM8iOiQ==
  /advanced-map-types/0.0.2:
    dependencies:
      '@types/node': 10.12.18
      tslib: 1.9.3
      typescript-miscellaneous: 0.1.1
    deprecated: Depends on wrong version of "convenient-typescript-utilities"
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-L6IJYPR2uFyt6W/R+1j70pG1M0KzkY9TpV0ySkJuhRAhOFxc2DZ3+0621eaivFoF5Yyc2+dQDwwpRUBwwa49iw==
  /ansi-styles/3.2.1:
    dependencies:
      color-convert: 1.9.3
    dev: false
    engines:
      node: '>=4'
    resolution:
      integrity: sha512-VT0ZI6kZRdTh8YyJw3SMbYm/u+NqfsAxEpWO0Pf9sq8/e94WxxOpPKx9FR1FlyCtOVDNOQ+8ntlqFxiRc+r5qA==
  /chalk/2.4.1:
    dependencies:
      ansi-styles: 3.2.1
      escape-string-regexp: 1.0.5
      supports-color: 5.5.0
    dev: false
    engines:
      node: '>=4'
    resolution:
      integrity: sha512-ObN6h1v2fTJSmUXoS3nMQ92LbDK9be4TV+6G+omQlGJFdcUX5heKi1LZ1YnRMIgwTLEj3E24bT6tYni50rlCfQ==
  /color-convert/1.9.3:
    dependencies:
      color-name: 1.1.3
    dev: false
    resolution:
      integrity: sha512-QfAUtd+vFdAtFQcC8CCyYt1fYWxSqAiK2cSD6zDB8N3cpsEBAvRxp9zOGg6G/SHHJYAT88/az/IuDGALsNVbGg==
  /color-name/1.1.3:
    dev: false
    resolution:
      integrity: sha1-p9BVi9icQveV3UIyj3QIMcpTvCU=
  /escape-string-regexp/1.0.5:
    dev: false
    engines:
      node: '>=0.8.0'
    resolution:
      integrity: sha1-G2HAViGQqN/2rjuyzwIAyhMLhtQ=
  /has-flag/3.0.0:
    dev: false
    engines:
      node: '>=4'
    resolution:
      integrity: sha1-tdRU3CGZriJWmfNGfloH87lVuv0=
  /supports-color/5.5.0:
    dependencies:
      has-flag: 3.0.0
    dev: false
    engines:
      node: '>=4'
    resolution:
      integrity: sha512-QjVjwdXIt408MIiAqCX4oUKsgU2EqAGzs2Ppkm4aQYbjm+ZEWEcW4SfFNTr4uMNZma0ey4f5lgLrkB0aX0QMow==
  /tslib/1.9.3:
    dev: false
    resolution:
      integrity: sha512-4krF8scpejhaOgqzBEcGM7yDIEfi0/8+8zDRZhNZZ2kjmHJ4hv3zCbQWxoJGz1iw5U0Jl0nma13xzHXcncMavQ==
  /typescript-miscellaneous/0.1.1:
    dependencies:
      '@types/node': 10.12.18
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
      typescript: '>= 3.0.0'
    resolution:
      integrity: sha512-dyGMilZPOL/qQMlJSWK+SbQyJpf6YZZPmPrbbdHGkAvGUvlhv4kGKTgddXWLxFWHTirR2I0VfD8mSwia3WLGtA==
registry: 'https://registry.npmjs.org/'
shrinkwrapMinorVersion: 9
shrinkwrapVersion: 3
specifiers:
  '@types/node': ^10.12.0
  '@types/which': ^1.3.1
  advanced-map-initialized: ^0.0.2
  chalk: ^2.4.1
  require-external-programs-db: ^0.0.6
  require-external-programs-lib: ^0.0.6
  tslib: ^1.9.3
