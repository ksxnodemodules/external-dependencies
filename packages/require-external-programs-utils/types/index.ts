import { CheckCommandsResult } from 'require-external-programs-db'

export type GroupRoleAssigner =
  (item: CheckCommandsResult.Item) => [string, string]
