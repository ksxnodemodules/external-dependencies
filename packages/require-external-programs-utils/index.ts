export * from './types'
export * from './enums'
export * from './classes'
export * from './functions'
import { examine } from './functions'
export default examine
