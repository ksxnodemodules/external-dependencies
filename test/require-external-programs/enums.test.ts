import { ExitStatusCodes } from 'require-external-programs-utils'

describe('ExitStatusCodes', () => {
  it('matches snapshot', () => {
    expect(ExitStatusCodes).toMatchSnapshot()
  })
})
