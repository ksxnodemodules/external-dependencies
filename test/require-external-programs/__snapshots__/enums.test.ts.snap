// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitStatusCodes matches snapshot 1`] = `
Object {
  "0": "Satisfied",
  "1": "Unsatisfied",
  "2": "Exception",
  "Exception": 2,
  "Satisfied": 0,
  "Unsatisfied": 1,
}
`;
